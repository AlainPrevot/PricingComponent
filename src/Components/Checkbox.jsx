import '../Sass/Checkbox.sass';
import PropTypes from 'prop-types';

export const Checkbox = ({ onMode }) => {
	return (
		<div className='container'>
			<span>Annually</span>
			<div className='checkbox'>
				<input type='checkbox' id='checkbox' onChange={() => onMode()} />
				<label htmlFor='checkbox'></label>
			</div>
			<span>Monthly</span>
		</div>
	);
};

Checkbox.propTypes = {
	onMode: PropTypes.func.isRequired,
};
