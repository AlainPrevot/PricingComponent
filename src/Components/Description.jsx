import PropTypes from 'prop-types'
import '../Sass/Description.sass'

export const Description = ({Storage, Users, Memory}) => {
  return (
    <div className='Card--Description border-top-bottom'>
        <span>{Storage} TB Storage</span>
        <span className='border-top-bottom'>{Users} Users Allowed</span>
        <span>Send up to {Memory} GB</span>
    </div>
  )
};

Description.proptypes = {
    Storage: PropTypes.number.isRequired,
    Users: PropTypes.number.isRequired,
    Memory: PropTypes.number.isRequired,
};

