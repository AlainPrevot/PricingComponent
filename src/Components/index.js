export * from './Card';
export * from './Checkbox';
export * from './ContainerCard';
export * from './Description';
export * from './Pricing';
export * from './Title';
export * from './TypePlan';