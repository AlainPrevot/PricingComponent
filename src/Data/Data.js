
const data = [
	{
		id: 1,
		title: 'Basic',
		pricing: 19.99,
		storage: 500,
		memory: 3,
		users: 2,
		theme: 'secundary',
	},
	{
		id: 2,
		title: 'Professional',
		pricing: 24.99,
		storage: 1,
		memory: 10,
		users: 5,
		theme: 'primary',
	},
	{
		id: 3,
		title: 'Master',
		pricing: 39.99,
		storage: 2,
		memory: 20,
		users: 10,
		theme: 'secundary',
	},
];

export default data