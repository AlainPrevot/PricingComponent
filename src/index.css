:root [data-theme='secundary'] {
	--background__card: hsl(0, 0%, 100%);
	--background__button: linear-gradient(
		145deg,
		hsl(236, 72%, 79%),
		hsl(237, 63%, 64%)
	);
	--font__primary: hsl(232, 13%, 33%);
	--font__button: hsl(0, 0%, 100%);
	--font__button--hover: hsl(237, 63%, 64%);
	--separation: hsl(234, 14%, 74%);
}
[data-theme='primary'] {
	--background__card: linear-gradient(
		145deg,
		hsl(236, 72%, 79%),
		hsl(237, 63%, 64%)
	);
	--background__button: hsl(0, 0%, 100%);
	--font__primary: hsl(0, 0%, 100%);
	--font__button: hsl(237, 63%, 64%);
	--font__button--hover: hsl(0, 0%, 100%);
	--separation: hsl(240, 78%, 98%);
}
