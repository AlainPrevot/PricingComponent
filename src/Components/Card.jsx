import PropTypes from 'prop-types';
import { Description } from './Description';
import { Pricing } from './Pricing';
import '../Sass/Card.sass';
import { usePlan } from '../Hooks/usePlan';

export const Card = ({
	title,
	pricing,
	storage,
	users,
	memory,
	theme,
	plane,
}) => {
	const { newPricing } = usePlan(pricing, plane);

	return (
		<div className='card' data-theme={theme}>
			<h2>{title}</h2>
			<Pricing value={newPricing} />
			<Description Storage={storage} Memory={memory} Users={users} />
			<button className='card--button'>LEARN MORE</button>
		</div>
	);
};

Card.propTypes = {
	title: PropTypes.string.isRequired,
	pricing: PropTypes.number.isRequired,
	storage: PropTypes.number.isRequired,
	memory: PropTypes.number.isRequired,
	users: PropTypes.number.isRequired,
	theme: PropTypes.string.isRequired,
	plane: PropTypes.string.isRequired,
};
