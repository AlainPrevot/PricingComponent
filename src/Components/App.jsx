import { Card } from './Card';
import '../Sass/App.sass';
import { Title } from './Title';
import { Checkbox } from './Checkbox';
import { useState } from 'react';

export const App = () => {
	const [plan, setPlan] = useState('monthly');

	const TypePlan = () => {
		setPlan(plan === 'monthly' ? 'Annually' : 'monthly');
	};

	return (
		<div
			style={{
				display: 'flex',
				alignItems: 'center',
				height: '76vh',
				// justifyContent: 'space-around',
				flexDirection: 'column',
			}}
			className='cuerpo'
		>
			<Title title={'Our Pricing'} />
			<Checkbox onMode={TypePlan} />
			<div className='prueba'>
				<Card
					title={'Basic'}
					pricing={19.99}
					storage={500}
					memory={3}
					users={2}
					theme={'secundary'}
					plane={plan}
				/>
				<Card
					title={'Professional'}
					pricing={24.99}
					storage={1}
					memory={10}
					users={5}
					theme={'primary'}
					plane={plan}
				/>
				<Card
					title={'Master'}
					pricing={39.99}
					storage={2}
					memory={20}
					users={10}
					theme={'secundary'}
					plane={plan}
				/>
			</div>
		</div>
	);
};
