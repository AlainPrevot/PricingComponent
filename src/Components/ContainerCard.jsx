import PropTypes from 'prop-types';
import { Card } from '../Components';
import '../Sass/ContainerCard.sass';
import data from '../Data/Data.js';

export const ContainerCard = ({ plan }) => {
	return (
		<div className='container--card'>
			{data.map(data => (
				<Card
					key={data.id}
					title={data.title}
					pricing={data.pricing}
					storage={data.storage}
					memory={data.memory}
					users={data.users}
					theme={data.theme}
					plane={plan}
				/>
			))}
		</div>
	);
};

ContainerCard.propTypes = {
	plan: PropTypes.string.isRequired,
};
