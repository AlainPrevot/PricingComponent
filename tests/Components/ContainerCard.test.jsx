import { render, screen } from '@testing-library/react';
import { ContainerCard } from '../../src/Components';
import { Data } from '../../src/Data/Data';

describe('Pruebas en el ContainerCard', () => {
	test('Debe de hacer match con el snapshot', () => {
		const { container } = render(<ContainerCard plan='monthly' />);

		expect(container).toMatchSnapshot();
	});

	test('Debe de mostrar un boton por cada Card', () => {
		render(<ContainerCard plan='monthly' />);

		expect(screen.getAllByRole('button').length).toBe(3); //Me aseguro que el botton de la card se dibuje las 3 veces que son necesarias
	});
});
