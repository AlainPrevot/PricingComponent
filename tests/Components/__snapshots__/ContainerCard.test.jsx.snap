// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el ContainerCard Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container--card"
  >
    <div
      class="card"
      data-theme="secundary"
    >
      <h2>
        Basic
      </h2>
      <div
        class="card--pricing"
      >
        <span>
          $
        </span>
        <h1>
          19.99
        </h1>
      </div>
      <ul
        class="card--description border-top-bottom"
      >
        <li>
          500
           
          GB
           Storage
        </li>
        <li
          class="border-top-bottom"
        >
          2
           Users Allowed
        </li>
        <li>
          Send up to 
          3
           GB
        </li>
      </ul>
      <button
        class="card--button"
      >
        LEARN MORE
      </button>
    </div>
    <div
      class="card"
      data-theme="primary"
    >
      <h2>
        Professional
      </h2>
      <div
        class="card--pricing"
      >
        <span>
          $
        </span>
        <h1>
          24.99
        </h1>
      </div>
      <ul
        class="card--description border-top-bottom"
      >
        <li>
          1
           
          TB
           Storage
        </li>
        <li
          class="border-top-bottom"
        >
          5
           Users Allowed
        </li>
        <li>
          Send up to 
          10
           GB
        </li>
      </ul>
      <button
        class="card--button"
      >
        LEARN MORE
      </button>
    </div>
    <div
      class="card"
      data-theme="secundary"
    >
      <h2>
        Master
      </h2>
      <div
        class="card--pricing"
      >
        <span>
          $
        </span>
        <h1>
          39.99
        </h1>
      </div>
      <ul
        class="card--description border-top-bottom"
      >
        <li>
          2
           
          TB
           Storage
        </li>
        <li
          class="border-top-bottom"
        >
          10
           Users Allowed
        </li>
        <li>
          Send up to 
          20
           GB
        </li>
      </ul>
      <button
        class="card--button"
      >
        LEARN MORE
      </button>
    </div>
  </div>
</div>
`;
